bundles/core/org.eclipse.smarthome.core.id/src/main/java/org/eclipse/smarthome/core/id/UUID.java

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright (c) 2014-2015 openHAB UG (haftungsbeschraenkt) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.core.id;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.List;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.smarthome.config.core.ConfigConstants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class provides a unique ID for the instance that can be used for identification, e.g. when
+ * integrating with external systems. The UUID is generated only once and written to the file system, so that it does
+ * not change over time.
+ *
+ * @author Kai Kreuzer - Initial contribution and API
+ */
+public class UUID {
+
+    static final private Logger logger = LoggerFactory.getLogger(UUID.class);
+
+    private static final String UUID_FILE_NAME = "uuid";
+
+    private static String uuid = null;
+
+    /**
+     * Retrieves a unified unique id, based on {@link java.util.UUID.randomUUID()}
+     *
+     * @return a UUID which identifies the instance
+     */
+    public static synchronized String get() {
+        if (uuid == null) {
+            File file = new File(ConfigConstants.getUserDataFolder() + File.separator + UUID_FILE_NAME);
+
+            if (!file.exists()) {
+                uuid = java.util.UUID.randomUUID().toString();
+                writeFile(file, uuid);
+            } else {
+                uuid = readFirstLine(file);
+                if (StringUtils.isNotEmpty(uuid)) {
+                    logger.debug("UUID file already exists at '{}' with content '{}'", file.getAbsolutePath(), uuid);
```

Comment:
Isn´t this the normal case after bundle restart or system restart? From the message it sounds a bit unexpected. Maybe something like "Found already existing UUID".